src/java.desktop/share/classes/com/sun/java/swing/SwingUtilities3.java

```diff
@@ -135,4 +142,76 @@ public static RepaintManager getDelegateRepaintManager(Component
         }
         return delegate;
     }
+
+    /**
+     * Used within border classes to add specific border implementation details.
+     */
+    @FunctionalInterface
+    public interface UnscaledBorderPainter {
+        void paintUnscaledBorder(Component c, Graphics g,
+                                 int w, int h,
+                                 double scaleFactor);
+    }
+
+    /**
+     * Performs common scaling transformation steps required for rendering
+     * the border correctly at different scales.
+     */
+    public static void paintBorder(Component c, Graphics g,
+                                   int x, int y,
+                                   int w, int h,
+                                   UnscaledBorderPainter painter) {
+
+        // Step 1: Reset Transform
+        AffineTransform at = null;
+        Stroke oldStk = null;
```

Comment:
```suggestion
        Stroke oldStroke = null;
```
Let's spell the word in full. It's used twice in the method and I see no reason to abbreviate it.